### PR TITLE
[APPS-2136] migrate search-datetime-range from moment to date-fns

### PR DESCRIPTION
**Pull request by vijayanssettu**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
moment dependency and moment adapter is present in the component

**What is the new behaviour?**
migration of dependency in the component from moment to date-fns

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://alfresco.atlassian.net/browse/APPS-2136

### CODE REVIEW

**Review by DenysVuika — APPROVED** [review]

File: lib/content-services/src/lib/search/components/search-datetime-range/search-datetime-range.component.ts [hunk 1]
```diff
@@ -133,8 +133,9 @@ export class SearchDatetimeRangeComponent implements SearchWidget, OnInit, OnDes
         if (isValid && this.id && this.context && this.settings && this.settings.field) {
```
Comment (rbahirsheth) [inline lib/content-services/src/lib/search/components/search-datetime-range/search-datetime-range.component.ts:133]: isValid parameter is conflicting with the method from date-fns. Please rename the parameter name.

Comment (vijayanssettu) [reply lib/content-services/src/lib/search/components/search-datetime-range/search-datetime-range.component.ts]: Added the rename the parameter name.

File: lib/content-services/src/lib/search/components/search-datetime-range/search-datetime-range.component.ts [hunk 2]
```diff
@@ -17,26 +17,27 @@
...
 import { UntypedFormControl, UntypedFormGroup, Validators } from '@angular/forms';
-import { UserPreferencesService, UserPreferenceValues } from '@alfresco/adf-core';
+import { DateFnsUtils, UserPreferencesService, UserPreferenceValues } from '@alfresco/adf-core';
 
 import { SearchWidget } from '../../models/search-widget.interface';
 import { SearchWidgetSettings } from '../../models/search-widget-settings.interface';
 import { SearchQueryBuilderService } from '../../services/search-query-builder.service';
 import { LiveErrorStateMatcher } from '../../forms/live-error-state-matcher';
-import { Moment } from 'moment';
+// import { Moment } from 'moment';
```
Comment (arohilaGL) [inline lib/content-services/src/lib/search/components/search-datetime-range/search-datetime-range.component.ts]: remove commented import

**Review by nikita-web-ua — APPROVED** [review]

File: lib/content-services/src/lib/search/components/search-datetime-range/search-datetime-range.component.ts [hunk 3]
```diff
@@ -129,12 +129,13 @@ export class SearchDatetimeRangeComponent implements SearchWidget, OnInit, OnDes
...
-    apply(model: { from: string; to: string }, isValid: boolean) {
-        if (isValid && this.id && this.context && this.settings && this.settings.field) {
+    apply(model: { from: string; to: string }, isDateValid: boolean) {
+        if (isDateValid && this.id && this.context && this.settings && this.settings.field) {
             this.isActive = true;
 
-            const start = moment.utc(model.from).startOf('minute').format();
-            const end = moment.utc(model.to).endOf('minute').format();
+            const start = format(startOfMinute(new Date(model.from)),`yyyy-MM-dd'T'HH:mm:ss'Z'`);
+            const end = format(endOfMinute(new Date(model.to)),`yyyy-MM-dd'T'HH:mm:ss'Z'`);
```
[the remark this answers is not in the document]
Comment (DenysVuika) [reply lib/content-services/src/lib/search/components/search-datetime-range/search-datetime-range.component.ts:137]: @vijayanssettu please address the comments

**Review by AleksanderSklorz — APPROVED** [review]

**Pull request closed without merging**